File: server/Models/FavoritedRecipe.cs
namespace allspice.Models;

public class FavoritedRecipe : RepoItem<int>
{
public int RecipeId { get; set; }
public string AccountId { get; set; }
}

public class UserFavoritedRecipe : Recipe
{
    public int RecipeId { get; set; }
    public string AccountId { get; set; }

    public int FavoriteId { get; set; }
}

File: client/src/models/FavoritedRecipe.js
import { Recipe } from "./Recipe.js";

export class FavoritedRecipe extends Recipe {
    constructor(data) {
        super(data)
        this.favoriteId = data.favoriteId
        this.accountId = data.accountId
        this.recipeId = data.recipeId
        this.isFavorited = data.isFavorited || true
    }
}